### PR TITLE
make sure to not-collapse different travel modes

### PR DESCRIPTION
**Pull request by MoKob**
# Issue

resolves https://github.com/Project-OSRM/osrm-backend/issues/2876

## Tasklist
 - [x] add regression / cucumber cases (see docs/testing.md)
 - [x] review
 - [x] adjust for for comments

### CODE REVIEW

File: features/guidance/collapse.feature [hunk 1]
```diff
@@ -792,3 +792,49 @@ Feature: Collapse
...
+            | bc    | primary |       |       | yes    |
+            | di    | service |       | serv  | yes    |
+            | ed    |         | ferry | ferry |        |
+            | gab   |         | ferry | ferry |        |
+            | kg    | primary |       | on    | yes    |
+            | ej    | primary |       | off   | yes    |
+
+        When I route I should get
+            | waypoints | route                   | turns                                                                          |
+            | k,j       | on,ferry,,ferry,off,off | depart,new name straight,continue uturn,turn straight,new name straight,arrive |
```
Comment (daniel-j-h) [inline features/guidance/collapse.feature]: The new name straight comes from the ferry way? To be honest I expected a notification here.

File: src/engine/guidance/post_processing.cpp [hunk 2]
```diff
@@ -472,16 +472,19 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
...
                     elongate(std::move(steps[one_back_index]), steps[next_step_index]);
                 invalidateStep(steps[next_step_index]); // will be skipped due to the
                                                         // continue statement at the
                                                         // beginning of this function
-            }
 
-            forwardStepSignage(steps[one_back_index], steps[two_back_index]);
+                forwardStepSignage(steps[one_back_index], steps[two_back_index]);
+            }
+            if( direct_u_turn  )
```
Comment (daniel-j-h) [inline src/engine/guidance/post_processing.cpp]: clang-format

File: features/guidance/collapse.feature (hunk 1 above)
Comment (MoKob) [inline features/guidance/collapse.feature]: Yeah. Thats something we have to improve on. It's a shortcoming from `getInstructionForObvious`. But thats a different issue (in the process of opening it)

File: src/engine/guidance/post_processing.cpp [hunk 3]
```diff
@@ -821,6 +827,7 @@ std::vector<RouteStep> collapseTurns(std::vector<RouteStep> steps)
         else if (isCollapsableInstruction(current_step.maneuver.instruction) &&
                  current_step.maneuver.instruction.type != TurnType::Suppressed &&
                  steps[getPreviousNameIndex(step_index)].name_id == current_step.name_id &&
+                 // canCollapseAll is also checking for compatible turns
```
Comment (daniel-j-h) [inline src/engine/guidance/post_processing.cpp]: What do you mean by this? compatible checks the travel mode, but what do you mean with compatible turns? from one travel mode to another I suppose?

Comment (MoKob) [inline src/engine/guidance/post_processing.cpp]: I am referring to the function `compatible` by name. I can also say steps if that is less confusing.

**Pull request merged**